### DISCUSSION TITLE
BUG - register - not confirm email - login - user get verified without code to exchange cookies

### DISCUSSION THREAD

**Discussion by nicitaacom** [supabase/supabase#20442] [Contribute to Supabase]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 
I not first day do auth with supabase so its not error in my application becasue I do everything my guide from YouTube

## Describe the bug
User get confirmed without code to exchange cookies

## To Reproduce

https://github.com/supabase/supabase/assets/39565703/0eccf2bf-2aed-47c9-9c9a-35172aa0ace0

I know its unsusal flow and common user will no do that but it work not as it should work

**Comment by saltcod** [comment]
If you could give us some more details @nicitaacom we might be able to point you in the right direction. This sounds like one to work through on Discord though.

**Comment by nicitaacom** [comment]
> This sounds like one to work through on Discord though.

Yes I'd like to show you this issue in discord (I mean supabase discord server)
Message me when its fit for you better fit for you 10:00-20:00 CET

**Comment by botxboom** [comment]
Hi @nicitaacom and @saltcod , have you started with this bug? I would like to work on this, if It is still open.

**Comment by nicitaacom** [comment]
> Hi @nicitaacom and @saltcod , have you started with this bug? I would like to work on this, if It is still open.

I didn't started with this bug
It would be nice if you fix this

**Comment by botxboom** [comment]
Sure, Will start with this then.

**Comment by nicitaacom** [comment]
> Sure, Will start with this then.

Ok thank you

Also please let us know if you figure it out - https://github.com/orgs/supabase/discussions/19306

**Comment by botxboom** [comment]
Hi @nicitaacom , I tried to replicate the same scenario as you have mentioned, but I am not getting any issue with PKCE. It is working fine.
for some reason, you are not getting `code verifier` from your url params
`const code = requestUrl.searchParams.get("code");`.

we can discuss more on discord.

**Comment by nicitaacom** [comment]
> we can discuss more on discord.

Yes I would like to
When its better for your to discuss more in discrod (on supabase server)?
For me its 10:00-20:00 CET

**Comment by botxboom** [comment]
> > we can discuss more on discord.
> 
> Yes I would like to When its better for your to discuss more in discrod (on supabase server)? For me its 10:00-20:00 CET

Yes Sure, link the supabase discussion here and we can discuss.

**Comment by nicitaacom** [comment]
> > > we can discuss more on discord.
> > 
> > 
> > Yes I would like to When its better for your to discuss more in discrod (on supabase server)? For me its 10:00-20:00 CET
> 
> Yes Sure, link the supabase discussion here and we can discuss.

I mean in voice here?
![image](https://github.com/supabase/supabase/assets/39565703/383c077e-c9b2-4294-b8f8-c8c100be4fd5)


When you are free to join voice channel and discuss about that?

**Comment by botxboom** [comment]
Hey @nicitaacom , we can discuss now.

**Comment by nicitaacom** [comment]
> Hey @nicitaacom , we can discuss now.

ok
join channel now :)

**Comment by botxboom** [comment]
Hi @nicitaacom , I couldn't stay active for past couple of days.
Let me know If you are still facing the same issue and want to discuss. Ping your availablility for today and we can join the channel.

**Comment by nicitaacom** [comment]
> Hi @nicitaacom , I couldn't stay active for past couple of days. Let me know If you are still facing the same issue and want to discuss. Ping your availablility for today and we can join the channel.

![image](https://github.com/supabase/supabase/assets/39565703/59e8ea39-0e1b-4561-bbea-c13d36edccbb)

**Comment by botxboom** [comment]
Okay, join the channel if possible

> > Hi @nicitaacom , I couldn't stay active for past couple of days. Let me know If you are still facing the same issue and want to discuss. Ping your availablility for today and we can join the channel.
> 
> ![image](https://private-user-images.githubusercontent.com/39565703/290981663-59e8ea39-0e1b-4561-bbea-c13d36edccbb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4MDExNTIsIm5iZiI6MTcwMjgwMDg1MiwicGF0aCI6Ii8zOTU2NTcwMy8yOTA5ODE2NjMtNTllOGVhMzktMGUxYi00NTYxLWJiZWEtYzEzZDM2ZWRjY2JiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE3VDA4MTQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3MmU2YzgyMzA1NGVjNzdmOTEzODI5ZDYzOWI0ZDU4NTczMzc1MGEwMmFiYzgwMjA0ZjhlYWY3YjVjOGRlY2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4XJWoEkGDrZ2WkfFH_MnmzFlc7u1rmMziP3q0c1w1uI)

Okay, join the channel and ping here.

**Comment by nicitaacom** [comment]
> Okay, join the channel if possible
> 
> > > Hi @nicitaacom , I couldn't stay active for past couple of days. Let me know If you are still facing the same issue and want to discuss. Ping your availablility for today and we can join the channel.
> > 
> > 
> > ![image](https://private-user-images.githubusercontent.com/39565703/290981663-59e8ea39-0e1b-4561-bbea-c13d36edccbb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4MDExNTIsIm5iZiI6MTcwMjgwMDg1MiwicGF0aCI6Ii8zOTU2NTcwMy8yOTA5ODE2NjMtNTllOGVhMzktMGUxYi00NTYxLWJiZWEtYzEzZDM2ZWRjY2JiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE3VDA4MTQxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3MmU2YzgyMzA1NGVjNzdmOTEzODI5ZDYzOWI0ZDU4NTczMzc1MGEwMmFiYzgwMjA0ZjhlYWY3YjVjOGRlY2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4XJWoEkGDrZ2WkfFH_MnmzFlc7u1rmMziP3q0c1w1uI)
> 
> Okay, join the channel and ping here.

I joined channel and ping you here

Ping like this @botxboom

**Comment by botxboom** [comment]
Hi @nicitaacom , I will be availble for full day today. I have joined the channel.

**Comment by nicitaacom** [comment]
> Hi @nicitaacom , I will be availble for full day today. I have joined the channel.

Seems like you AFK

![Uploading image.png…]()

**Comment by nicitaacom** [comment]
@botxboom thank you that you will look into it
Let me know if you need additional info about this issue
As you saw if you verify email from that comes from 'resend' function from supabase (and user have no confirmed email) it throws error if you handle error like me - or it just set no cookies
